Tests failing with error in babel-standalone

While working on https://github.com/babel/babel/pull/6381 I've got error after run `make test` locally. 

```
make test
./node_modules/.bin/eslint scripts packages codemods *.js --format=codeframe --rulesdir="./eslint_rules"
./scripts/test.sh
/Users/rusinov/my-projects/babel/packages/babel-standalone/babel.js:80
throw new Error("Module build failed: TypeError: [BABEL] /Users/rusinov/my-projects/babel/packages/babel-standalone/src/index.js: Cannot read property 'loose' of undefined (While processing: \"base$2\")\n    at _default (/Users/rusinov/my-projects/babel/packages/babel-plugin-transform-optional-chaining/lib/index.js:12:31)\n    at /Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:239:12\n    at cachedFunction (/Users/rusinov/my-projects/babel/packages/babel-core/lib/config/caching.js:48:17)\n    at loadPluginDescriptor (/Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:269:28)\n    at /Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:56:14\n    at Array.map (<anonymous>)\n    at OptionManager.mergeOptions (/Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:55:34)\n    at /Users/rusinov/my-projects/babel/packages/b

Error: Module build failed: TypeError: [BABEL] /Users/rusinov/my-projects/babel/packages/babel-standalone/src/index.js: Cannot read property 'loose' of undefined (While processing: "base$2")
    at _default (/Users/rusinov/my-projects/babel/packages/babel-plugin-transform-optional-chaining/lib/index.js:12:31)
    at /Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:239:12
    at cachedFunction (/Users/rusinov/my-projects/babel/packages/babel-core/lib/config/caching.js:48:17)
    at loadPluginDescriptor (/Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:269:28)
    at /Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:56:14
    at Array.map (<anonymous>)
    at OptionManager.mergeOptions (/Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:55:34)
    at /Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:83:15
    at Array.forEach (<anonymous>)
    at OptionManager.mergeOptions (/Users/rusinov/my-projects/babel/packages/babel-core/lib/config/option-manager.js:82:15)
    at Object.<anonymous> (/Users/rusinov/my-projects/babel/packages/babel-standalone/babel.js:80:7)
    at __webpack_require__ (/Users/rusinov/my-projects/babel/packages/babel-standalone/babel.js:30:30)
    at /Users/rusinov/my-projects/babel/packages/babel-standalone/babel.js:73:18
    at /Users/rusinov/my-projects/babel/packages/babel-standalone/babel.js:76:10
    at webpackUniversalModuleDefinition (/Users/rusinov/my-projects/babel/packages/babel-standalone/babel.js:3:20)
    at Object.<anonymous> (/Users/rusinov/my-projects/babel/packages/babel-standalone/babel.js:10:3)
    at Module._compile (module.js:624:30)
    at Module._compile (/Users/rusinov/my-projects/babel/node_modules/pirates/lib/index.js:88:24)
    at Module._extensions..js (module.js:635:10)
    at Object.newLoader [as .js] (/Users/rusinov/my-projects/babel/node_modules/pirates/lib/index.js:93:7)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/rusinov/my-projects/babel/packages/babel-standalone/test/babel.js:2:15)
    at Module._compile (module.js:624:30)
    at Module._compile (/Users/rusinov/my-projects/babel/node_modules/pirates/lib/index.js:88:24)
    at Module._extensions..js (module.js:635:10)
    at Object.newLoader [as .js] (/Users/rusinov/my-projects/babel/node_modules/pirates/lib/index.js:93:7)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at /Users/rusinov/my-projects/babel/node_modules/mocha/lib/mocha.js:230:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/Users/rusinov/my-projects/babel/node_modules/mocha/lib/mocha.js:227:14)
    at Mocha.run (/Users/rusinov/my-projects/babel/node_modules/mocha/lib/mocha.js:513:10)
    at Object.<anonymous> (/Users/rusinov/my-projects/babel/node_modules/mocha/bin/_mocha:480:18)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Function.Module.runMain (module.js:665:10)
    at startup (bootstrap_node.js:201:16)
    at bootstrap_node.js:626:3
make: *** [test-only] Error 1
```


| software         | version(s)
| ---------------- | -------
| babel-standalone            |  7.0.0-beta.2
| node             | 8.5.0
| npm              | 5.4.1
| Operating System | macOS Sierra 10.12.6

--- a/packages/babel-core/src/config/option-manager.js
+++ b/packages/babel-core/src/config/option-manager.js
@@ -220,6 +220,11 @@ type BasicDescriptor = {
   loc: string,
 };
 
+type LoadedDescriptor = {
+  value: {},
+  descriptor: BasicDescriptor,
+};
+
 /**
  * Load and validate the given config into a set of options, plugins, and presets.
  */
@@ -282,38 +287,40 @@ const loadConfig = makeWeakCache((config): {
 /**
  * Load a generic plugin/preset from the given descriptor loaded from the config object.
  */
-const loadDescriptor = makeWeakCache((descriptor, cache) => {
-  if (typeof descriptor.value !== "function") {
-    return { value: descriptor.value, descriptor };
-  }
-  const { value, options } = descriptor;
-
-  const api = Object.assign(Object.create(context), {
-    cache,
-    env: () => cache.using(() => getEnv()),
-  });
-
-  let item;
-  try {
-    item = value(api, options, { dirname: descriptor.dirname });
-  } catch (e) {
-    if (descriptor.alias) {
-      e.message += ` (While processing: ${JSON.stringify(descriptor.alias)})`;
+const loadDescriptor = makeWeakCache(
+  (descriptor: BasicDescriptor, cache): LoadedDescriptor => {
+    if (typeof descriptor.value !== "function") {
+      return { value: descriptor.value, descriptor };
     }
-    throw e;
-  }
+    const { value, options } = descriptor;
 
-  if (!item || typeof item !== "object") {
-    throw new Error("Plugin/Preset did not return an object.");
-  }
+    const api = Object.assign(Object.create(context), {
+      cache,
+      env: () => cache.using(() => getEnv()),
+    });
 
-  return { value: item, descriptor };
-});
+    let item;
+    try {
+      item = value(api, options, { dirname: descriptor.dirname });
+    } catch (e) {
+      if (descriptor.alias) {
+        e.message += ` (While processing: ${JSON.stringify(descriptor.alias)})`;
+      }
+      throw e;
+    }
+
+    if (!item || typeof item !== "object") {
+      throw new Error("Plugin/Preset did not return an object.");
+    }
+
+    return { value: item, descriptor };
+  },
+);
 
 /**
  * Instantiate a plugin for the given descriptor, returning the plugin/options pair.
  */
-function loadPluginDescriptor(descriptor: BasicDescriptor) {
+function loadPluginDescriptor(descriptor: BasicDescriptor): Plugin {
   if (descriptor.value instanceof Plugin) {
     if (descriptor.options) {
       throw new Error(
@@ -328,7 +335,7 @@ function loadPluginDescriptor(descriptor: BasicDescriptor) {
 }
 
 const instantiatePlugin = makeWeakCache(
-  ({ value: pluginObj, descriptor }, cache) => {
+  ({ value: pluginObj, descriptor }: LoadedDescriptor, cache): Plugin => {
     Object.keys(pluginObj).forEach(key => {
       if (!ALLOWED_PLUGIN_KEYS.has(key)) {
         throw new Error(
@@ -391,15 +398,17 @@ const loadPresetDescriptor = (descriptor: BasicDescriptor): MergeOptions => {
   return instantiatePreset(loadDescriptor(descriptor));
 };
 
-const instantiatePreset = makeWeakCache(({ value, descriptor }) => {
-  return {
-    type: "preset",
-    options: value,
-    alias: descriptor.alias,
-    loc: descriptor.loc,
-    dirname: descriptor.dirname,
-  };
-});
+const instantiatePreset = makeWeakCache(
+  ({ value, descriptor }: LoadedDescriptor): MergeOptions => {
+    return {
+      type: "preset",
+      options: value,
+      alias: descriptor.alias,
+      loc: descriptor.loc,
+      dirname: descriptor.dirname,
+    };
+  },
+);
 
 /**
  * Validate and return the options object for the config.

--- a/packages/babel-core/src/config/option-manager.js
+++ b/packages/babel-core/src/config/option-manager.js
@@ -289,24 +289,25 @@ const loadConfig = makeWeakCache((config): {
  */
 const loadDescriptor = makeWeakCache(
   (descriptor: BasicDescriptor, cache): LoadedDescriptor => {
-    if (typeof descriptor.value !== "function") {
-      return { value: descriptor.value, descriptor };
-    }
     const { value, options } = descriptor;
 
-    const api = Object.assign(Object.create(context), {
-      cache,
-      env: () => cache.using(() => getEnv()),
-    });
+    let item = value;
+    if (typeof value === "function") {
+      const api = Object.assign(Object.create(context), {
+        cache,
+        env: () => cache.using(() => getEnv()),
+      });
 
-    let item;
-    try {
-      item = value(api, options, { dirname: descriptor.dirname });
-    } catch (e) {
-      if (descriptor.alias) {
-        e.message += ` (While processing: ${JSON.stringify(descriptor.alias)})`;
+      try {
+        item = value(api, options, { dirname: descriptor.dirname });
+      } catch (e) {
+        if (descriptor.alias) {
+          e.message += ` (While processing: ${JSON.stringify(
+            descriptor.alias,
+          )})`;
+        }
+        throw e;
       }
-      throw e;
     }
 
     if (!item || typeof item !== "object") {

--- a/packages/babel-core/src/config/option-manager.js
+++ b/packages/babel-core/src/config/option-manager.js
@@ -222,7 +222,10 @@ type BasicDescriptor = {
 
 type LoadedDescriptor = {
   value: {},
-  descriptor: BasicDescriptor,
+  options: ?{},
+  dirname: string,
+  alias: string,
+  loc: string,
 };
 
 /**
@@ -289,7 +292,7 @@ const loadConfig = makeWeakCache((config): {
  */
 const loadDescriptor = makeWeakCache(
   (descriptor: BasicDescriptor, cache): LoadedDescriptor => {
-    const { value, options } = descriptor;
+    const { value, options, dirname, alias, loc } = descriptor;
 
     let item = value;
     if (typeof value === "function") {
@@ -314,7 +317,7 @@ const loadDescriptor = makeWeakCache(
       throw new Error("Plugin/Preset did not return an object.");
     }
 
-    return { value: item, descriptor };
+    return { value: item, options, descriptor, dirname, alias, loc };
   },
 );
 
@@ -336,11 +339,14 @@ function loadPluginDescriptor(descriptor: BasicDescriptor): Plugin {
 }
 
 const instantiatePlugin = makeWeakCache(
-  ({ value: pluginObj, descriptor }: LoadedDescriptor, cache): Plugin => {
+  (
+    { value: pluginObj, options, dirname, alias, loc }: LoadedDescriptor,
+    cache,
+  ): Plugin => {
     Object.keys(pluginObj).forEach(key => {
       if (!ALLOWED_PLUGIN_KEYS.has(key)) {
         throw new Error(
-          `Plugin ${descriptor.alias} provided an invalid property of ${key}`,
+          `Plugin ${alias} provided an invalid property of ${key}`,
         );
       }
     });
@@ -364,11 +370,11 @@ const instantiatePlugin = makeWeakCache(
     let inherits;
     if (plugin.inherits) {
       inheritsDescriptor = {
-        alias: `${descriptor.loc}$inherits`,
-        loc: descriptor.loc,
+        alias: `${loc}$inherits`,
+        loc,
         value: plugin.inherits,
-        options: descriptor.options,
-        dirname: descriptor.dirname,
+        options,
+        dirname,
       };
 
       // If the inherited plugin changes, reinstantiate this plugin.
@@ -388,7 +394,7 @@ const instantiatePlugin = makeWeakCache(
       ]);
     }
 
-    return new Plugin(plugin, descriptor.options, descriptor.alias);
+    return new Plugin(plugin, options, alias);
   },
 );
 
@@ -400,13 +406,13 @@ const loadPresetDescriptor = (descriptor: BasicDescriptor): MergeOptions => {
 };
 
 const instantiatePreset = makeWeakCache(
-  ({ value, descriptor }: LoadedDescriptor): MergeOptions => {
+  ({ value, dirname, alias, loc }: LoadedDescriptor): MergeOptions => {
     return {
       type: "preset",
       options: value,
-      alias: descriptor.alias,
-      loc: descriptor.loc,
-      dirname: descriptor.dirname,
+      alias,
+      loc,
+      dirname,
     };
   },
 );

--- a/packages/babel-core/src/config/option-manager.js
+++ b/packages/babel-core/src/config/option-manager.js
@@ -222,7 +222,7 @@ type BasicDescriptor = {
 
 type LoadedDescriptor = {
   value: {},
-  options: {} | void,
+  options: {},
   dirname: string,
   alias: string,
   loc: string,
@@ -292,7 +292,7 @@ const loadConfig = makeWeakCache((config): {
  */
 const loadDescriptor = makeWeakCache(
   (descriptor: BasicDescriptor, cache): LoadedDescriptor => {
-    const { value, options, dirname, alias, loc } = descriptor;
+    const { value, options = {}, dirname, alias, loc } = descriptor;
 
     let item = value;
     if (typeof value === "function") {

--- a/packages/babel-core/src/config/option-manager.js
+++ b/packages/babel-core/src/config/option-manager.js
@@ -214,7 +214,7 @@ class OptionManager {
 
 type BasicDescriptor = {
   value: {} | Function,
-  options: ?{},
+  options: {} | void,
   dirname: string,
   alias: string,
   loc: string,
@@ -222,7 +222,7 @@ type BasicDescriptor = {
 
 type LoadedDescriptor = {
   value: {},
-  options: ?{},
+  options: {} | void,
   dirname: string,
   alias: string,
   loc: string,
@@ -520,7 +520,7 @@ function normalizePair(
 ): {
   filepath: string | null,
   value: {} | Function,
-  options: ?{},
+  options: {} | void,
 } {
   let options;
   let value = pair;
@@ -562,6 +562,7 @@ function normalizePair(
       "Plugin/Preset options must be an object, null, or undefined",
     );
   }
+  options = options || undefined;
 
   return { filepath, value, options };
 }

--- a/packages/babel-core/src/config/plugin.js
+++ b/packages/babel-core/src/config/plugin.js
@@ -7,9 +7,9 @@ export default class Plugin {
   pre: ?Function;
   visitor: ?{};
 
-  options: {} | void;
+  options: {};
 
-  constructor(plugin: {}, options: {} | void, key?: string) {
+  constructor(plugin: {}, options: {}, key?: string) {
     if (plugin.name != null && typeof plugin.name !== "string") {
       throw new Error("Plugin .name must be a string, null, or undefined");
     }

--- a/packages/babel-core/src/config/plugin.js
+++ b/packages/babel-core/src/config/plugin.js
@@ -9,7 +9,7 @@ export default class Plugin {
 
   options: {} | void;
 
-  constructor(plugin: {}, options: ?{}, key?: string) {
+  constructor(plugin: {}, options: {} | void, key?: string) {
     if (plugin.name != null && typeof plugin.name !== "string") {
       throw new Error("Plugin .name must be a string, null, or undefined");
     }
@@ -37,6 +37,6 @@ export default class Plugin {
     this.post = plugin.post;
     this.pre = plugin.pre;
     this.visitor = plugin.visitor;
-    this.options = options || undefined;
+    this.options = options;
   }
 }

